src/constants.rs

```
@@ -1,26 +1,95 @@
-    "ETH/USD" => "ethereum",
-    "LUSD/USD" => "liquity-usd",
-    "WBTC/USD" => "wrapped-bitcoin",
-    "DAI/USD" => "dai",
-    "USDC/USD" => "usd-coin",
-    "USDT/USD" => "tether",
-    "WSTETH/USD" => "wrapped-steth",
-    "LORDS/USD" => "lords",
-    "STRK/USD" => "starknet",
-    "ZEND/USD" => "zklend-2",
-    "NSTR/USD" => "nostra",
-};
+// Define the Coin struct to store the id and symbol of the coin.
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Coin {
+    id: String,
+    symbol: String,
+}
+
+// We already have the Link for the CoinGecko API, so we can use it to fetch the data.
+// We will use the CoinGecko API to fetch the data.
+// Example: https://api.coingecko.com/api/v3/coins/markets?vs_currency=usd&order=market_cap_desc&per_page=100&page=1
+// We will fetch the data for the first 100 coins.
+// here is the data how its look like:
+// [
+//     {
+//       "id": "bitcoin",
+//       "symbol": "btc",
+//       "name": "Bitcoin",
+//       "image": "https://coin-images.coingecko.com/coins/images/1/large/bitcoin.png?1696501400",
+//       "current_price": 100390,
+//       ...
+//     },
+//     ...
+// ]
+#[allow(dead_code)]
+async fn get_coingecko_mappings() -> Result<HashMap<String, String>, Box<dyn Error>> {
+    let client = reqwest::Client::new();
+    let mut mappings = HashMap::new();
+    let mut page = 1;
+
+    loop {
+        let url = format!(
+            "https://api.coingecko.com/api/v3/coins/markets?vs_currency=usd&order=market_cap_desc&per_page=100&page={}",
+            page
+        );
+
+        let response = client
+            .get(&url)
+            .header("User-Agent", "Crypto Data Fetcher")
+            .send()
+            .await?;
+
+        if !response.status().is_success() {
+            return Err(format!("API request failed with status: {}", response.status()).into());
+        }
+
+        let coins: Vec<Coin> = response.json().await?;
+        if coins.is_empty() {
+            break;
```

we should probably return an error here as it's definitely not normal behavior

---

Make sense.